splinterd/api/static/openapi.yml

```diff
@@ -327,6 +327,145 @@ paths:
         500:
           description: Internal service error
 
+  /user/notifications:
+    get:
+      description: Retrieve a list of all notifications
+      responses:
+        200:
+          description: Notifications have been retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                $ref: '#/components/schemas/Notification'
+        403:
+          description: User not authorized to access this endpoint
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+      description: Publish a list of notifications
+      responses:
+        200:
+          description: Notifications have been published
+        400:
+          description: Malformed notifications list
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        500:
+          description: Internal service error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  /user/users/{user_id}/notifications:
+    get:
+      description: Retrieve a list of notifications for a given user from the notifications database
```

Comment:
I don't think that the user-specific notification should include the full list of recipients.

Reply:
Is there any reason that it shouldn't?

Reply:
It seems problematic to expose all of the uids to all requesters even if they arent authorized to view all notifications. That information is also not useful to Canopy in general when displaying notifications.

Reply:
updated